core/lib/config/src/configs/da_client/avail.rs

```diff
@@ -1,16 +1,40 @@
+use std::str::FromStr;
+
+use secrecy::ExposeSecret as _;
+use secrecy::Secret;
 use serde::Deserialize;
 use zksync_basic_types::seed_phrase::SeedPhrase;
 
+#[derive(Clone, Debug)]
+pub struct GasRelayAPIKey(pub Secret<String>);
+
+impl PartialEq for GasRelayAPIKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.expose_secret().eq(other.0.expose_secret())
+    }
+}
+
+impl FromStr for GasRelayAPIKey {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(GasRelayAPIKey(s.parse()?))
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Deserialize)]
 pub struct AvailConfig {
-    pub api_node_url: String,
+    pub api_node_url: Option<String>,
```

Comment:
it seems like this conditional presence of some values in the config is quite confusing (especially for someone who would read the code and not know that usual client and gas relay are mutually exclusive), it would be much more understandable if we had 2 separate configs: AvailConfig and AvailGasRelayConfig, and then you can think maybe it makes sense to group them in enum (but I don't have an opinion here, up to you)

Reply:
Fixed, closing.